src/sysupdate/sysupdated.c

```diff
@@ -0,0 +1,1498 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-bus.h"
+
+#include "bus-get-properties.h"
+#include "bus-label.h"
+#include "bus-log-control-api.h"
+#include "bus-polkit.h"
+#include "common-signal.h"
+#include "discover-image.h"
+#include "env-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "json.h"
+#include "main-func.h"
+#include "memfd-util.h"
+#include "missing_capability.h"
+#include "mkdir-label.h"
+#include "os-util.h"
+#include "process-util.h"
+#include "service-util.h"
+#include "signal-util.h"
+#include "socket-util.h"
+#include "string-table.h"
+#include "user-util.h"
+
+typedef struct Manager {
+        sd_event *event;
+        sd_bus *bus;
+
+        Hashmap *targets;
+
+        uint32_t current_job_id;
+        Hashmap *jobs;
+
+        Hashmap *polkit_registry;
+
+        int notify_fd;
+        sd_event_source *notify_event_source;
+} Manager;
+
+typedef enum TargetClass {
+        /* These should try to match ImageClass from src/basic/os-util.h */
+        TARGET_MACHINE = IMAGE_MACHINE,
+        TARGET_PORTABLE = IMAGE_PORTABLE,
+        TARGET_SYSEXT = IMAGE_SYSEXT,
+        TARGET_CONFEXT = IMAGE_CONFEXT,
```

Comment:
maybe add `assert_cc(_IMAGE_CLASS_MAX == 4)` here, so that anyone updating that other enum wil get a compile failure if they don't update it here.